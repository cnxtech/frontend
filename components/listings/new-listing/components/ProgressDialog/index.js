import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Button from '@emcasa/ui-dom/components/Button'
import Row from '@emcasa/ui-dom/components/Row'
import Col from '@emcasa/ui-dom/components/Col'
import Text from '@emcasa/ui-dom/components/Text'
import Container from 'components/listings/new-listing/shared/Container'
import { BUTTON_WIDTH } from 'components/listings/new-listing/styles'

class ProgressDialog extends Component {
  render() {
    return (
      <Container>
        <Col width={[1,null,null,1/2]}>
          <Row>
            <Text>Você já estava cadastrando seu imóvel da {this.props.address}. Gostaria de continuar de onde parou?</Text>
          </Row>
          <Row justifyContent="space-between">
            <Col>
              <Button style={{width: BUTTON_WIDTH}} fluid onClick={this.props.onReset}>Não</Button>
            </Col>
            <Col>
              <Button style={{width: BUTTON_WIDTH}} fluid active onClick={this.props.onResume}>Sim</Button>
            </Col>
          </Row>
        </Col>
      </Container>
    )
  }

  static propTypes = {
    address: PropTypes.string.isRequired,
    onReset: PropTypes.func.isRequired,
    onResume: PropTypes.func.isRequired
  }

  static defaultProps = {
    address: null,
    onReset: null,
    onResume: null
  }
}

export default ProgressDialog
