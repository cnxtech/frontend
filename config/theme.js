import theme from '@emcasa/ui'
import {fas} from '@fortawesome/pro-solid-svg-icons'
import {fal} from '@fortawesome/pro-light-svg-icons'
import icons from './icons'

export default {
  ...theme,
  icons: {emcasa: icons, default: fas, light: fal}
}
