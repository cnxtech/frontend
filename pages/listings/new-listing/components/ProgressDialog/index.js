import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Button from '@emcasa/ui-dom/components/Button'
import Row from '@emcasa/ui-dom/components/Row'
import Col from '@emcasa/ui-dom/components/Col'
import Text from '@emcasa/ui-dom/components/Text'

class ProgressDialog extends Component {
  render() {
    return (
      <Row m={4} justifyContent="center">
        <Col width={[1, 1/2]}>
          <Row>
            <Text>Você já estava cadastrando seu imóvel da {this.props.address}. Gostaria de continuar de onde parou?</Text>
          </Row>
          <Row justifyContent="space-between">
            <Col width={[1/3, 5/12]}>
              <Button fluid onClick={this.props.onReset}>Não</Button>
            </Col>
            <Col width={[1/3, 5/12]}>
              <Button fluid active onClick={this.props.onResume}>Sim</Button>
            </Col>
          </Row>
        </Col>
      </Row>
    )
  }

  static propTypes = {
    address: PropTypes.string.isRequired,
    onReset: PropTypes.func.isRequired,
    onResume: PropTypes.func.isRequired
  }

  static defaultProps = {
    address: null,
    onReset: null,
    onResume: null
  }
}

export default ProgressDialog
