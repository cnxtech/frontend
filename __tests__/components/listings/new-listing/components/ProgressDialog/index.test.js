import React from 'react'
import ProgressDialog from 'components/listings/new-listing/components/ProgressDialog'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import theme from 'config/theme'
import { ThemeProvider } from 'styled-components'

describe('<ProgressDialog/>', () => {
  it('should render component', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <ProgressDialog
          address="Rua tal"
          onReset={() => {}}
          onResume={() => {}}
        />
      </ThemeProvider>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
