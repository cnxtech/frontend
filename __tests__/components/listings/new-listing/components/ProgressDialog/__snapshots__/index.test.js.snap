// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressDialog/> should render component 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  outline: none;
  display: block;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #e0e6ed;
  background-color: #ffffff;
  border-radius: 4px;
  border-width: 1px;
  width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  height: 40px;
  color: #3c4858;
  font-family: Rubik;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-style: solid;
}

.c6 {
  outline: none;
  display: block;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #ff2b7f;
  background-color: #ff2b7f;
  border-radius: 4px;
  border-width: 1px;
  width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  height: 40px;
  color: #ffffff;
  font-family: Rubik;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-style: solid;
}

.c1 {
  width: 100%;
}

.c3 {
  -webkit-font-smoothing: antialiased;
  font-family: Rubik;
  color: #3c4858;
  font-size: 18px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 80px 20px 20px 20px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media screen and (min-width:780px) {
  .c1 {
    width: 50%;
  }
}

@media (min-width:480px) and (max-width:779px) {
  .c0 {
    padding-top: 180px;
    width: calc(100vw - 220px);
  }
}

<div
  className="c0"
>
  <div
    className="c1"
    width={
      Array [
        1,
        null,
        null,
        0.5,
      ]
    }
  >
    <div
      className="c2"
    >
      <p
        className="c3"
        color="dark"
        fontSize="default"
      >
        Você já estava cadastrando seu imóvel da 
        Rua tal
        . Gostaria de continuar de onde parou?
      </p>
    </div>
    <div
      className="c4"
    >
      <div
        className=""
      >
        <button
          className="c5"
          fontSize="default"
          height="medium"
          onClick={[Function]}
          style={
            Object {
              "width": 130,
            }
          }
        >
          Não
        </button>
      </div>
      <div
        className=""
      >
        <button
          className="c6"
          fontSize="default"
          height="medium"
          onClick={[Function]}
          style={
            Object {
              "width": 130,
            }
          }
        >
          Sim
        </button>
      </div>
    </div>
  </div>
</div>
`;
